x-pack/legacy/plugins/ml/public/components/kql_filter_bar/utils.js

```diff
@@ -3,11 +3,11 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-
-
-import { getAutocompleteProvider } from 'ui/autocomplete_providers';
+import { npSetup } from 'ui/new_platform';
 import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
 
+const getAutocompleteProvider = language => npSetup.plugins.data.autocomplete.getProvider(language);
```

Comment:
Get rid of `getAutocompleteProvider` and use `data.autocomplete.getProvider` directly

Reply:
const { data } = npSetup.plugins;

...

data.autocomplete.getProvider(language);

Reply:
Not sure that it's a good idea. 
Please see the following case: 
1) `file1.ts`
```
import {npSetup} from 'np';
  
const { data } = npSetup.plugins;
  
export const  CONSTANT = 'someVar'; 
export const  getProvider = () => { 
    data.autocomplete.getProvider(language);
}
```

2) `file2.ts`

```
import { CONSTANT } from './file1'; 

export const foo() => { 
   console.log(CONSTANT);
}

```

Now please try to write some tests for `file2.ts`.  Unfortunately without adding mocks of data plugin you cannot do it.

Reply:
Yeah, that was just a suggestion.
I would just replace `getAutocompleteProvider` with `npStart.plugins.data.autocomplete.getProvider`.

But as you wish on this one.